### PR TITLE
Support other form envs

### PR DESCRIPTION
**Pull request by ogp-weeloong**
# Problem
We need to support non-prod FormSG envs

# Solution
If the user provides a non-prod FormSG URL as their form ID, we will switch to the appropriate SDK and API key to perform form operations such as connection, etc.

# Tests
- Updated old unit tests and added new unit tests
- Check that I can connect to staging forms.
- Check that staging form connections have a `[STAGING]` prefix.
- Regression test: check that I can still connect to prod forms
- Regression test: check that prod form connections don't have a prefix

# Deploy Notes
**Parameter Store Changes**
- Renamed `plumber-<ENVIRONMENT>-formsg-api-key` to `plumber-formsg-api-key`, since we don't maintain separate keys for each env anyway.

**Env vars**
- `FORMSG_STAGING_API_KEY`:  Our API key for staging env. We will need one key env var for each FormSG env.
- `FORMSG_UAT_API_KEY`:  Our API key for staging env. We will need one key env var for each FormSG env.

### CODE REVIEW

**Comment by ogp-weeloong** [comment]
* **#666** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/666?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ogp-weeloong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/src/config/app-env-vars/formsg.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const formSgConfig = Object.freeze({
+  apiKeys: {
+    prod: process.env.FORMSG_API_KEY,
```
Comment (ogp-weeloong) [inline packages/backend/src/config/app-env-vars/formsg.ts:3]: Getting more and more clunky to manage these env vars. Thinking to move to a different approach where we query these secrets from our DB instead (see this abandoned [PR](https://github.com/opengovsg/plumber/pull/616) for example - I created an `app_metadata` table)

File: packages/backend/src/apps/formsg/auth/decrypt-form-response.ts [hunk 2]
```diff
@@ -54,13 +50,15 @@ export async function decryptFormResponse(
     return false
   }
 
+  const formSgSdk = getSdk(parseFormEnv($))
```
Comment (m0nggh) [inline packages/backend/src/apps/formsg/auth/decrypt-form-response.ts:53]:
```suggestion
  const formsgSdk = getSdk(parseFormEnv($))
```
super nit seems like formsg should just all be small letters? up to you I'm just putting it out here

File: packages/backend/src/apps/formsg/auth/verify-credentials.ts [hunk 3]
```diff
@@ -46,12 +52,15 @@ export const verifyFormCreds = async (
     throw new Error('Form is not a storage mode form')
   }
 
-  if (!formsgSdk().crypto.valid(publicKey, secretKey)) {
+  const formSdk = getSdk(env)
```
Comment (m0nggh) [inline packages/backend/src/apps/formsg/auth/verify-credentials.ts]:
```suggestion
  const formsgSdk = getSdk(env)
```
can make the naming consistent here again?

File: packages/backend/src/apps/formsg/common/form-env.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+export const SUPPORTED_FORM_ENVS = ['prod', 'staging'] as const
+export type FormEnv = (typeof SUPPORTED_FORM_ENVS)[number]
+
+export function parseFormIdAsUrl(rawUrl: string): URL | null {
+  try {
+    const url = new URL(
+      rawUrl.startsWith('http') ? rawUrl : `https://${rawUrl}`,
+    )
+
+    return url.hostname === 'form.gov.sg' ||
```
Comment (m0nggh) [inline packages/backend/src/apps/formsg/common/form-env.ts:23]: can i check why is the first condition check necessary ah? doesn't the 2nd one alr cover the first

File: packages/backend/src/config/app-env-vars/formsg.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export const formSgConfig = Object.freeze({
```
Comment (m0nggh) [inline packages/backend/src/config/app-env-vars/formsg.ts]:
```suggestion
export const formsgConfig = Object.freeze({
```
same thing for consistency sake if you decide to change the naming

File: packages/backend/src/apps/formsg/common/form-env.ts (hunk 4 above)
Comment (pregnantboy) [reply packages/backend/src/apps/formsg/common/form-env.ts:23]: 2nd check has the `.` in front, which is needed cos we dont wanna match deform.gov.sg

**Review by pregnantboy — APPROVED** [review]
lgtm!

File: packages/backend/src/apps/formsg/auth/verify-credentials.ts (hunk 3 above)
Comment (ogp-weeloong) [reply packages/backend/src/apps/formsg/auth/verify-credentials.ts]: good call, thanks!

File: packages/backend/src/config/app-env-vars/formsg.ts (hunk 5 above)
Comment (ogp-weeloong) [reply packages/backend/src/config/app-env-vars/formsg.ts]: good call, done!

**Pull request merged**